Applications/FileIO/SWMM/SWMMInterface.cpp

```diff
@@ -0,0 +1,1248 @@
+    GeoLib::Polyline* p (nullptr);
+    getline(in, line);
+    while (!isSectionFinished(line))
+    {
+        if (isCommentLine(line))
+        {
+            getline(in, line);
+            continue;
+        }
+
+        std::vector<std::string> split_str (BaseLib::splitString(line));
+        if (split_str.size() != 3)
+        {
+            ERR ("Polygon format not recognised.");
+            return false;
+        }
+
+        // if a new polygon starts, add the old one to the vector
+        if (split_str[0] != polygon_name)
+        {
+            if (p != nullptr)
+                lines.push_back(p);
+
+            polygon_name = split_str[0];
+            p = new GeoLib::Polyline(points);
+            ply_names.push_back(polygon_name);
+        }
+
+        double const x = BaseLib::str2number<double>(split_str[1]);
+        double const y = BaseLib::str2number<double>(split_str[2]);
+        GeoLib::Point* pnt = new GeoLib::Point(x, y, 0, id);
+        points.push_back(pnt);
+        p->addPoint(points.size()-1);
+        pnt_names.push_back("");
+        id++;
+        getline(in, line);
+    }
+
+    // when the section is finished, add the last polygon
+    if (p != nullptr)
+        lines.push_back(p);
+
+    return true;
+}
+
+bool SwmmInterface::addPointElevation(std::ifstream &in,
+    std::vector<GeoLib::Point*> &points, std::map<std::string,
+    std::size_t> const& name_id_map)
+{
+    std::string line;
+    getline(in, line);
+    while (!isSectionFinished(line))
+    {
+        if (isCommentLine(line))
+        {
+            getline(in, line);
+            continue;
+        }
+
+        std::vector<std::string> const split_str (BaseLib::splitString(line));
+        // Junctions = 6, Outfalls = 4, Storage = 8
+        if (split_str.size() < 4)
+        {
+            ERR ("Format not recognised.");
+            return false;
+        }
+        std::string const current_name (split_str[0]);
+        auto const it = name_id_map.find(current_name);
+        if (it == name_id_map.end())
+        {
+            ERR ("SwmmInterface::addPointElevation(): Name %s not found coordinates map.", current_name.c_str());
+            return false;
+        }
+        std::size_t const id = it->second;
+        (*points[id])[2] = BaseLib::str2number<double>(split_str[1]);
+        getline(in, line);
+    }
+    return true;
+}
+
+bool SwmmInterface::readLinksAsPolylines(std::ifstream &in,
+    std::vector<GeoLib::Polyline*> &lines, std::vector<std::string> &line_names,
+    std::vector<GeoLib::Point*> const& points, std::map<std::string, std::size_t> const& point_names)
+{
+    std::string line;
+    getline(in, line);
+    while (!isSectionFinished(line))
+    {
+        if (isCommentLine(line))
+        {
+            getline(in, line);
+            continue;
+        }
+
+        std::vector<std::string> const split_str (BaseLib::splitString(line));
+        // Conduits = 9, Pumps = 7, Weirs = 8
+        if (split_str.size() < 7)
```

Comment:
The lines look like that
```
;;               Inlet            Outlet                      Manning    Inlet      Outlet     Init.      Max.      
;;Name           Node             Node             Length     N          Offset     Offset     Flow       Flow      
;;-------------- ---------------- ---------------- ---------- ---------- ---------- ---------- ---------- ----------
Drosselstrecke   343              zuKA             44.48951562 0.013      0          0          0          40        
```
and for various entries I have no idea what they mean.